Core/Services/ReviewCommentService.cs

```diff
@@ -0,0 +1,88 @@
+        private readonly IReviewRepository _reviewRepository;
+        private readonly IGetterService<Book> _bookGetter;
+        public ReviewCommentService(IReviewCommentRepository reviewCommentRepository, ILoggedUserProvider loggedUserProvider,
+            IReviewRepository reviewRepository, IGetterService<Book> bookGetter)
+        {
+            _reviewCommentRepository = reviewCommentRepository;
+            _loggedUserProvider = loggedUserProvider;
+            _reviewRepository = reviewRepository;
+            _bookGetter = bookGetter;
+        }
+        public async Task<ServiceResponse> AddReviewComment(ReviewCommentRequest comment)
+        {
+            if (!(await _reviewCommentRepository.ReviewExists(comment.ReviewId)))
+                return ServiceResponse.Error($"Review you're trying to post a comment for doesn't exist.", HttpStatusCode.BadRequest);
+
+            var userId = _loggedUserProvider.GetUserId();
+            if (await _reviewCommentRepository.CheckCommentCount(comment.ReviewId, userId))
+                return ServiceResponse.Error($"You can post only {_reviewCommentRepository.MaxCommentPerReview} " +
+                    $"comments per single review", HttpStatusCode.BadRequest);
+
+            if (await _reviewCommentRepository.CheckCommentsDate(comment.ReviewId, userId))
+                return ServiceResponse.Error($"You can post only {_reviewCommentRepository.MaxCommentPerHourPerReview} " +
+                    $"comments per hour on a single review. Try again later", HttpStatusCode.BadRequest);
+
+            return ServiceResponse<ReviewCommentDto>.Success(await _reviewCommentRepository.CreateReviewComment(comment),
+                "Comment posted.", HttpStatusCode.Created);
+        }
+
+        public async Task<ServiceResponse> GetComment(int reviewCommentId)
+        {
+            var reviewComment = await _reviewCommentRepository.GetComment(reviewCommentId);
+            if (reviewComment != null)
+                return ServiceResponse<ReviewCommentDto>.Success(reviewComment, "Comment retrieved.");
+
+            return ServiceResponse.Error("Comment not found.", HttpStatusCode.NotFound);
+        }
+
+        public async Task<ServiceResponse> GetComments(int? reviewId, Guid? userId, bool currentUser)
+        {
+            if (reviewId != null && !(await _reviewCommentRepository.ReviewExists((int)reviewId)))
+                return ServiceResponse.Error($"Review doesn't exist.", HttpStatusCode.BadRequest);
+
+            if (reviewId == null && userId == null && currentUser != true)
+                return ServiceResponse<IEnumerable<ReviewCommentDto>>.Success(_reviewCommentRepository.GetComments(),
+                    $"All comments retrieved.");
+
+            var message = await CreateMessage(reviewId, userId, currentUser);
+            if (currentUser) userId = _loggedUserProvider.GetUserId();
+            var comments = _reviewCommentRepository.GetComments(reviewId, userId);
```

Comment:
I would swap this logic a little. 
First, we can check if we can assign current user to userId and then we can build the message.
After that we can retrieve comments (the GetComments method with reviewId = null and userId = null will anyway return results of base GetComments).
Finally, we just return result, so the upper `return` statement isn't required